docs/code_gen_options.rst

```diff
@@ -218,4 +218,26 @@ always true, so the check is omitted.
 This also means that, whenever the length of an array is accessed using '.length', it is replaced with a constant.
 
 Note that this optimization does not cover every case. When an array is passed
-as a function argument, for instance, the length is unknown.
\ No newline at end of file
+as a function argument, for instance, the length is unknown.
+
+Debugging Options
+=================
+
+It may be desirable to have access to debug information regarding the contract execution. 
+How-ever, this might lead to less ideal code as well as much higher gas usage. Hence it is disabled by default,
```

Comment:
```suggestion
However, this might lead to more instructions as well as higher gas usage. Hence, it is disabled by default,
```
I think `less ideal` sounds awkward, so this suggestion is optional. Please, be sure to remove the dash from `how-ever` and place a comma after `hence`.